### PR TITLE
Shape analysis notebook

### PR DESCRIPTION
**Pull request by alebrigant**
This PR
- creates notebook on use of square root velocity metric and its quotient metric
- adds the quotient distance to `QuotientSRVMetric`

### CODE REVIEW

File: geomstats/geometry/discrete_curves.py [hunk 1]
```diff
@@ -1270,3 +1273,39 @@ def horizontal_path(t):
             return horizontal_path
 
         return horizontal_path
+
+    def quotient_dist(self, point_a, point_b, n_times=20, threshold=1e-3):
```
Comment (ninamiolane) [inline geomstats/geometry/discrete_curves.py:1277]: For clarity for non-experts, maybe specify that this the quotient with respect to "parameterizations", i.e. with respect to sampling along the curve?

File: tests/tests_geomstats/test_discrete_curves.py [hunk 2]
```diff
@@ -283,25 +282,25 @@ def test_is_tangent_to_tangent(self):
         result = self.space_curves_in_sphere_2d.is_tangent(tangent_vec, point)
         self.assertTrue(gs.all(result))
 
-    # @geomstats.tests.np_autograd_and_torch_only
-    # def test_projection_closed_curves(self):
-    #     """Test that projecting the projection returns the projection
+    @geomstats.tests.np_autograd_and_torch_only
+    def test_projection_closed_curves(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_discrete_curves.py:286]: 🎉

File: tests/tests_geomstats/test_discrete_curves.py [hunk 3]
```diff
@@ -550,3 +549,24 @@ def test_horizontal_geodesic(self):
...
+            gs.stack(
+                (
+                    gs.zeros(self.n_sampling_points),
+                    gs.zeros(self.n_sampling_points),
+                    gs.linspace(1.0, 0.5, self.n_sampling_points),
+                )
+            )
+        )
+        result = self.quotient_srv_metric_r3.quotient_dist(curve_a_resampled, curve_b)
+        expected = self.quotient_srv_metric_r3.quotient_dist(self.curve_a, curve_b)
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_discrete_curves.py:571]: Why not calling `quotient_dist` just `dist`? This variable name seems redundant: "quotient_metric.quotient_dist".

**Review by ninamiolane — COMMENTED** [review]
Awesome!! 🎉  This notebook was very much needed, thank you.

**Comment by alebrigant** [comment]
> View / edit / reply to [this conversation](https://app.reviewnb.com/geomstats/geomstats/pull/1165/discussion/) on ReviewNB
> 
> _ninamiolane commented on 2021-09-21T14:45:02Z_
> ----------------------------------------------------------------
> 
> Typo: This is achieved.
> 
> Maybe specify that the `pre_shape.py` module can be used to remove translations and rotations by using the "align" method, and refer to
> 
> -https://github.com/geomstats/geomstats/blob/master/notebooks/usecase_optic_nerve_heads_analysis_in_kendall_shape_space.ipynb for details?
> 
> Maybe also refer to Florent's notebook in the ICLR 2021 challenge, about the use of the SRVMetric as an example on cell data? https://github.com/geomstats/challenge-iclr-2021/blob/main/Florent-Michel__Shape-Analysis-of-Cancer-Cells/submission_cell_shape_analysis.ipynb
> 
> We could also import part of his notebook as a `usecase_*` notebook in geomstats.

Thanks for the very good suggestions !

**Pull request merged**